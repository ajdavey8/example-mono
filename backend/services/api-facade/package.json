{
  "name": "test",
  "private": true,
  "dependencies": {
    "@andrew-org-test/backend-common": "1.2.2"
  },
  "devDependencies": {
    "@tsconfig/node18": "^2.0.0",
    "typescript": "^5.0.4"
  },
  "version": "3.0.0"
}
