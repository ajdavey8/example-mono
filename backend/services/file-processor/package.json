{
  "name": "file-processro",
  "private": true
}
