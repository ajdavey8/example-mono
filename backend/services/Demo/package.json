{
  "name": "@shieldpay/Demo",
  "version": "1.0.0",
  "author": "Shieldpay",
  "description": "",
  "private": true,
  "scripts": {
    "build": "yarn sst build --stage $STAGE",
    "compile": "tsc --build",
    "deploy": "DEPLOY_REGION=eu-west-2 yarn sst deploy --stage $STAGE --verbose && DEPLOY_REGION=eu-west-1 yarn sst deploy --stage $STAGE --verbose",
    "dev": "../../../scripts/sst/run-sst-dev.sh backend/services/api-facade",
    "lint": "eslint .",
    "lint:fix": "eslint . --fix",
    "remove-deployment": "DEPLOY_REGION=eu-west-2 yarn sst remove --stage $STAGE --verbose && DEPLOY_REGION=eu-west-1 yarn sst remove --stage $STAGE --verbose",
    "test": "yarn jest --runInBand --silent __tests__/test/*",
    "test:coverage": "yarn jest --coverage --runInBand --silent __tests__/test/*",
    "typecheck": "tsc --noEmit"
  },
  "dependencies": {
    "@shieldpay/common": "workspace:*",
    "@shieldpay/http": "workspace:*",
    "@shieldpay/logger": "workspace:*",
    "@shieldpay/shared": "workspace:*",
    "@shieldpay/util": "workspace:*",
    "pino": "8.11.0"
  },
  "devDependencies": {
    "@aws-sdk/client-dynamodb": "3.315.0",
    "@shieldpay/eslint-config-backend": "workspace:*",
    "@shieldpay/prettier-config-backend": "workspace:*",
    "@shieldpay/var-check": "workspace:*",
    "@tsconfig/node18": "2.0.0",
    "@types/aws-lambda": "8.10.111",
    "@types/jest": "29.1.2",
    "aws-cdk-lib": "2.72.1",
    "constructs": "10.1.156",
    "eslint": "8.36.0",
    "eslint-config-prettier": "8.7.0",
    "eslint-plugin-jest": "27.2.1",
    "jest": "29.5.0",
    "nock": "13.3.0",
    "prettier": "2.8.4",
    "sst": "2.5.3",
    "ts-jest": "29.0.5",
    "typescript": "5.0.4"
  }
}
