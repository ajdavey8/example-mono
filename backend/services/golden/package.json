{
  "name": "@shieldpay/golden-service",
  "author": "Shieldpay",
  "version": "1.0.0",
  "description": "Optimus BE-golden service",
  "scripts": {
    "clean:dependencies": "rm -rf node_modules",
    "build": "STAGE=${STAGE:-dev} npx cdk synth",
    "compile": "tsc --build",
    "deploy": "STAGE=${STAGE:-dev} npx cdk deploy --require-approval never --all",
    "lint:fix": "eslint . --fix",
    "lint": "eslint .",
    "remove-deployment": "STAGE=${STAGE:-dev} cdk destroy --force --all",
    "test:coverage": "yarn jest --coverage --runInBand --silent",
    "test": "yarn jest --runInBand --silent",
    "typecheck": "tsc --noEmit",
    "migrate:up:test": "knex migrate:latest --knexfile ./src/knex/knexfile.ts --env testing",
    "migrate:down:test": "knex migrate:rollback --knexfile ./src/knex/knexfile.ts --env testing",
    "migrate:aws": "yarn sls invoke -f knex-run-migrations --stage ${STAGE:-dev} --log"
  },
  "dependencies": {
    "@middy/core": "4.3.0",
    "@middy/http-cors": "4.3.0",
    "@middy/http-error-handler": "4.3.0",
    "@middy/http-json-body-parser": "4.3.0",
    "constructs": "10.0.113",
    "knex": "2.4.2",
    "pino": "8.8.0"
  },
  "devDependencies": {
    "aws-cdk-lib": "2.74.0",
    "eslint": "8.36.0",
    "eslint-config-prettier": "8.7.0",
    "eslint-plugin-jest": "27.2.1",
    "jest": "29.5.0",
    "nock": "13.3.0",
    "ts-jest": "29.0.5",
    "typescript": "5.0.4"
  }
}
